### PR TITLE
Support <ARRAY-SIZE-HANDLING>

### PR DESCRIPTION
**Pull request by andredami**

### CODE REVIEW

**Comment by andredami** [comment]
[[autosar] Support <ARRAY-SIZE-HANDLING>](https://trello.com/c/zK2hmeyf/798-autosar-support-array-size-handling)

File: autosar/parser/datatype_parser.py [hunk 1]
```diff
@@ -324,13 +326,13 @@ def parseApplicationArrayDataType(self, xmlRoot, parent = None):
     @parseElementUUID
     def parseApplicationArrayElement(self, xmlRoot):
         assert (xmlRoot.tag == 'ELEMENT')
-        (typeRef, arraySize, sizeHandling, sizeSemantics) = (None, None, None, None)
+        (typeRef, arraySize, sizeHandling, sizeSemantics, arraySizeHandling) = (None, None, None, None, None)
```
Comment (marco-rabozzi) [inline autosar/parser/datatype_parser.py:329]: arraySizeHandling looks unused, I think you can revert this line change

File: autosar/parser/datatype_parser.py [hunk 2]
```diff
@@ -404,6 +406,14 @@ def _parseModeRequestTypeMapXML(self, xmlRoot):
                 raise NotImplementedError(xmlElem.tag)
         return autosar.datatype.ModeRequestTypeMap(modeDeclarationGroupRef, implementationDataTypeRef)
 
+    def parseArraySizeHandling(self, xmlRoot):
+        assert (xmlRoot.tag == 'ARRAY-SIZE-HANDLING')
+        text = self.parseTextNode(xmlRoot)
+        for literal in autosar.datatype.ArraySizeHandlingEnum:
+            if literal.value == text:
+                return literal
+        raise ValueError(f"Invalid value for ArraySizeHandling field: {text}")
```
Comment (marco-rabozzi) [inline autosar/parser/datatype_parser.py:415]: maybe enclose text in quotes so that if the xml contains invalid spaces we can detect them: ```'{text}'```

**Review by marco-rabozzi — CHANGES_REQUESTED** [review]
minors

**Review by marco-rabozzi — APPROVED** [review]
LGTM

**Pull request merged**